test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -183,20 +191,20 @@ instance Semigroup WalletState where
 
 instance IsOurs WalletState where
     isOurs addr s@(WalletState ours discovered) =
-        if addr `elem` ours then
-            (True, WalletState ours (Set.insert addr discovered))
+        if (ShowFmt addr) `elem` ours then
+            (True, WalletState ours (Set.insert (ShowFmt addr) discovered))
         else
             (False, s)
 
 instance Arbitrary WalletState where
-    shrink (WalletState ours _) =
-        [ WalletState (Set.fromList ours') mempty
-        | ours' <- shrinkList pure (Set.toList ours)
-        ]
+    shrink = genericShrink
```

Comment:
So this was the reason for the tests hanging :| ... The shrinker was actually shrinking to itself, which caused QC to loop ... and loop ... and loop... I've changed for a generic shrinker which seems to do the job here as long as we make the output slightly nicer through the `ShowFmt`

Reply:
It is / was stuck because the previous shrinker would in some cases yield the same value as the one being shrinked (which will cause QuickCheck to enter a shrinking-loop).

Reply:
yes that makes sense - I didn't look at the old version.

I am still puzzled why ghci tells that `Generic` isnt defined for `Set Int` :|

Reply:
There's no instance of `Generic` for `Set`. It can be derived, but it has to be explicit I suppose.